pkg/services/anonymous/anonimpl/api/api.go

```diff
@@ -0,0 +1,95 @@
+package api
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/grafana/grafana/pkg/api/dtos"
+	"github.com/grafana/grafana/pkg/api/response"
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/anonymous"
+	"github.com/grafana/grafana/pkg/services/anonymous/anonimpl/anonstore"
+	contextmodel "github.com/grafana/grafana/pkg/services/contexthandler/model"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/util"
+)
+
+type AnonDeviceServiceAPI struct {
+	cfg            *setting.Cfg
+	store          anonstore.AnonStore
+	accesscontrol  accesscontrol.AccessControl
+	RouterRegister routing.RouteRegister
+	log            log.Logger
+}
+
+func NewAnonDeviceServiceAPI(
+	cfg *setting.Cfg,
+	anonstore anonstore.AnonStore,
+	accesscontrol accesscontrol.AccessControl,
+	routerRegister routing.RouteRegister,
+) *AnonDeviceServiceAPI {
+	return &AnonDeviceServiceAPI{
+		cfg:            cfg,
+		store:          anonstore,
+		accesscontrol:  accesscontrol,
+		RouterRegister: routerRegister,
+		log:            log.New("anon.api"),
+	}
+}
+
+func (api *AnonDeviceServiceAPI) RegisterAPIEndpoints() {
+	auth := accesscontrol.Middleware(api.accesscontrol)
+	api.RouterRegister.Group("/api/anonymous", func(anonRoutes routing.RouteRegister) {
+		anonRoutes.Get("/stats", auth(accesscontrol.EvalPermission(accesscontrol.ActionServerStatsRead)), routing.Wrap(api.CountDevices))
+		anonRoutes.Get("/devices", auth(accesscontrol.EvalPermission(accesscontrol.ActionUsersRead)), routing.Wrap(api.ListDevices))
+	})
+}
+
+func (api *AnonDeviceServiceAPI) ListDevices(c *contextmodel.ReqContext) response.Response {
+	fromTime := time.Now().Add(-anonymous.ThirtyDays)
+	toTime := time.Now().Add(time.Minute)
+
+	devices, err := api.store.ListDevices(c.Req.Context(), &fromTime, &toTime)
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, "Failed to list devices", err)
```

Comment:
```suggestion
		return response.ErrOrFallback(http.StatusInternalServerError, "Failed to list devices", err)
```

Reply:
Right now I think the only error we could get is that the sql query faild and that should be a 500. But if we in the future have other error and are using `errutil.Error` this would correctly set the status code and response message

Reply:
thx for correcting and explaining ⚡